internal/component/otelcol/internal/scheduler/scheduler.go

```diff
@@ -37,13 +37,33 @@ type Scheduler struct {
 	schedMut        sync.Mutex
 	schedComponents []otelcomponent.Component // Most recently created components
 	host            otelcomponent.Host
+	running         bool
+
+	// onPause is called when scheduler is making changes to running components.
+	onPause func()
+	// onResume is called when scheduler is done making changes to running components.
+	onResume func()
 }
 
+// TODO: Delete this function? I don't think it's used anywhere.
```

Comment:
I would be ok to delete it because it may actually lead to mistakes but you can check with Piotr because he decided to create a different one

Reply:
I just realised we need to keep it - it's used by `otelcol.auth.`, `otelcol.extension.`, and `otelcol.receiver.`.